src/DataTable/tests/DataTable.test.jsx

```diff
@@ -128,47 +123,50 @@ describe('<DataTable />', () => {
   beforeEach(() => {
     jest.clearAllMocks();
   });
+
   it('displays the empty table component if empty', () => {
-    const wrapper = mount(<DataTableWrapper {...props} data={[]} />);
-    expect(wrapper.find(EmptyTable).length).toEqual(1);
+    render(<DataTableWrapper {...props} data={[]} />);
+    expect(screen.getByText('No results found')).toBeInTheDocument();
   });
+
   it('accepts an empty table component', () => {
-    const wrapper = mount(<DataTableWrapper {...props} data={[]} EmptyTableComponent={EmptyTest} />);
-    expect(wrapper.find(EmptyTable).length).toEqual(0);
-    expect(wrapper.find(EmptyTest).length).toEqual(1);
+    render(<DataTableWrapper {...props} data={[]} EmptyTableComponent={EmptyTest} />);
+    expect(screen.queryByText('No results found')).not.toBeInTheDocument();
+    expect(screen.getByText(emptyTestText)).toBeInTheDocument();
   });
+
   it('displays a control bar', () => {
-    const wrapper = mount(<DataTableWrapper {...props} />);
-    const controlBar = wrapper.find(TableControlBar);
-    expect(controlBar.length).toEqual(1);
-    expect(controlBar.text()).toEqual('Showing 7 of 7.');
+    render(<DataTableWrapper {...props} />);
+    expect(screen.getByTestId('control-bar-id')).toBeInTheDocument();
+    expect(screen.getAllByText('Showing 7 of 7.')[0]).toBeInTheDocument();
   });
+
   it('displays a table', () => {
-    const wrapper = mount(<DataTableWrapper {...props} />);
-    const table = wrapper.find(Table);
-    expect(table.length).toEqual(1);
-    expect(table.find('th').length).toEqual(3);
-    expect(table.find('tr').length).toEqual(8);
+    render(<DataTableWrapper {...props} />);
+    expect(screen.getAllByRole('columnheader')).toHaveLength(3); // Assuming 3 columns
+    expect(screen.getAllByRole('row')).toHaveLength(8); // Assuming 8 rows
```

Comment:
Instead of assuming you could take these value from the props you pass to the table, i.e
```js
expect(screen.getAllByRole('columnheader')).toHaveLength(props.columns.length);
expect(screen.getAllByRole('row')).toHaveLength(props.data.length + 1); // (need + 1 to include header row)
```

this way tests won't break if someone adds another row / column.

Same approach should probably work for other tests in this file.